### DISCUSSION TITLE
networktocode.nautobot.inventory - query_filters - multiple roles values not working

### DISCUSSION THREAD

**Discussion by n0l0g0n** [nautobot/nautobot#1748] [Q&A] [answered]
Hi,

I have a network of about 1200 devices,  Where the most devices are access devices.
However, i don't need to return all devices, i want the devices from only the selected roles to return which is about 50+ devices.

At this moment I tried a lot of variations within query_filters: and this works fine with just one role.
So I am facing the challenge to have either a full-blown query returning all devices or an incomplete  subset based on roles.
I am running
Collection             Version
---------------------- -------
networktocode.nautobot 3.3.1 

I checked the discussions here and I don't find anything like my challenge.

When i look at the documentation at https://nautobot-ansible.readthedocs.io/en/latest/plugins/inventory_inventory.html
query_filters list / elements=string | List of parameters passed to the query string for both devices and VMs (Multiple values may be separated by commas) Default: []
-- | --

I might have interpreted this wrong but it would suggest i can use multiple values separated by commas
Below are a few attempts and yes running dr or dcs as a single role yields results, I tried all the variations on presenting multiple values but none were succesful
```
query_filters:
  - role:
    - dr
    - dcs
[WARNING]:  * Failed to parse /home/thijn/Development/zerotouchprovisioning/inventories/nautobot_inventory_ztp.yml with auto
plugin: {"role":["Select a valid choice. ['dr', 'dcs'] is not one of the available choices."]}
```
```
query_filters:
  - role: dr dcs
[WARNING]:  * Failed to parse /home/thijn/Development/zerotouchprovisioning/inventories/nautobot_inventory_ztp.yml with auto
plugin: {"role":["Select a valid choice. dr dcs is not one of the available choices."]}
```
```
query_filters:
  - role: dr, dcs
[WARNING]:  * Failed to parse /home/thijn/Development/zerotouchprovisioning/inventories/nautobot_inventory_ztp.yml with auto
plugin: {"role":["Select a valid choice. dr, dcs is not one of the available choices."]}
```

So, I assume I am not the first one to have this challenge.. how would I achieve a single inventory run where I can yield results showing all devices in multiple roles?
or.. is it just not possible?

thank you
Kinds regards,
Thijn

**Comment by itdependsnetworks** [comment, marked as answer]
If you review one of the examples you will see the pattern
```
query_filters:
  - role: server
  - tag: web
  - tag: production
```
so in your case, I think you are looking for 
```
query_filters:
  - role: dr
  - role: dcs
```

**Reply by n0l0g0n** [reply]
How did i miss that!!! thanx! that works
